packages/vm/src/modules/in_memory_cache.rs

```diff
@@ -1,39 +1,48 @@
 use clru::CLruCache;
-use wasmer::Module;
+use wasmer::{Module, Store};
 
 use crate::{Checksum, Size, VmResult};
 
 const ESTIMATED_MODULE_SIZE: Size = Size::mebi(10);
 
 /// An in-memory module cache
 pub struct InMemoryCache {
-    lru: CLruCache<Checksum, Module>,
+    artifacts: CLruCache<Checksum, Vec<u8>>,
 }
 
 impl InMemoryCache {
     /// Creates a new cache with the given size (in bytes)
     pub fn new(size: Size) -> Self {
         let max_entries = size.0 / ESTIMATED_MODULE_SIZE.0;
         InMemoryCache {
-            lru: CLruCache::new(max_entries),
+            artifacts: CLruCache::new(max_entries),
         }
     }
 
     pub fn store(&mut self, checksum: &Checksum, module: Module) -> VmResult<()> {
-        self.lru.put(*checksum, module);
+        let serialized_artifact = module.serialize()?;
+        self.artifacts.put(*checksum, serialized_artifact);
         Ok(())
     }
 
-    pub fn load(&mut self, checksum: &Checksum) -> VmResult<Option<&Module>> {
-        let optional = self.lru.get(checksum);
-        Ok(optional)
+    /// Looks up a module in the cache and takes its artifact and
+    /// creates a new module from store and artifact.
+    pub fn load(&mut self, checksum: &Checksum, store: &Store) -> VmResult<Option<Module>> {
```

Comment:
Isn't this more expensive/slow than just returning the in-memory module. Why serialize/deserialize it?

Reply:
It is. The problem is you need to reset the store (module = artifact + store) and we don't have an API for that: https://github.com/wasmerio/wasmer/issues/1933

Reply:
Thus the artifact store PR. Now it all makes sense